src/storage/src/hummock/local_version/local_hummock_version.rs

```diff
@@ -0,0 +1,331 @@
+// Copyright 2023 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::{HashMap, HashSet};
+use std::sync::Arc;
+
+use itertools::Itertools;
+use risingwave_common::catalog::TableId;
+use risingwave_hummock_sdk::compaction_group::hummock_version_ext::{
+    add_new_sub_level, summarize_group_deltas, GroupDeltasSummary, HummockLevelsExt,
+};
+use risingwave_hummock_sdk::compaction_group::{StateTableId, StaticCompactionGroupId};
+use risingwave_hummock_sdk::CompactionGroupId;
+use risingwave_pb::hummock::hummock_version::Levels;
+use risingwave_pb::hummock::{
+    CompactionConfig, HummockVersion, HummockVersionDelta, LevelType, SstableInfo,
+};
+
+use crate::hummock::Sstable;
+
+#[derive(Clone, Debug)]
+pub struct SubLevelCache {
+    pub sub_level_id: u64,
+    pub data: Vec<Arc<Sstable>>,
+}
+
+impl SubLevelCache {
+    pub fn new(sub_level_id: u64) -> Self {
+        Self {
+            sub_level_id,
+            data: vec![],
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct LocalGroup {
+    pub raw_group_meta: Levels,
+    pub base_level_cache: Vec<Vec<Arc<Sstable>>>,
+    pub sub_level_cache: Vec<SubLevelCache>,
```

Comment:
Yes. I would add some debug_assert in next PR. Because this PR has not support preload meta. We can check this property every time after we load or drop meta data.